pkg/querier/tripperware/roundtrip.go

```diff
@@ -156,7 +156,8 @@ func NewQueryTripperware(
 				now := time.Now()
 				userStr := tenant.JoinTenantIDs(tenantIDs)
 				activeUsers.UpdateUserTimestamp(userStr, now)
-				queriesPerTenant.WithLabelValues(op, userStr).Inc()
+				source := getSource(r.Header.Get("User-Agent"))
+				queriesPerTenant.WithLabelValues(op, userStr, source).Inc()
```

Comment:
We probably shouldn't have the user-agent as a possible label value.

Reply:
If it can be set by the user, then there's a possibility of cardinality explosion.

Reply:
Good call out. I think `source` is good enough and we don't need the user agent here

Reply:
The label values would be `ruler` or `api`. Don't we?

Reply:
Nvm ignore my comment. We are not using user agent value directly as label value.

Reply:
Can we add a test here to make sure the user agent is either ruler or api to ensure we don't allow values other than those?

Reply:
Thanks, update it.